cpp/modmesh/simd/neon/neon.hpp

```diff
@@ -115,6 +115,13 @@ const T * check_between(T const * start, T const * end, T const & min_val, T con
     T const * ret = NULL;
 
     T const * ptr = start;
+
+    // Check if array is large enough for SIMD processing
```

Comment:
This prevents wrong counting when the number of elements is less than a SIMD vector width for `check_between()` (vector mode).

I am curious whether this issue is also in other vector-mode calculating functions?  @KHLee529 what do you think?